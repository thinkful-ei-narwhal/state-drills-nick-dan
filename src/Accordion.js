import React from 'react';

export default class Accordion extends React.Component {
  static defaultProps = {
    sections: [
      { title: "Thing 1", content: "Lorem ipsum dolor sit amet." },
      { title: "Thing 2", content: "This should appear and the others go away." },
      { title: "Thing 3", content: "I said GO AWAY!" },
    ]
  };

  state = {
    pressed: null
  }

  handleClick = index => {
    this.setState({ pressed: index })
  };

  render() {
    const { sections } = this.props;
    const { pressed } = this.state;
    return (
      <ul className="container">
        {sections.map((section, index) => {
          return (<li key={index} >
            <button onClick={() => this.handleClick(index)}>{section.title}</button>
            {(pressed === index) && <p>{section.content}</p>}
          </li>)
        })}
      </ul>
    );
  }
}