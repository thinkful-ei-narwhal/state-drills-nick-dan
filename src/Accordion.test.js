import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Accordion from './Accordion';

describe(`Accordian Component`, () => {
  it('renders without errors', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Accordion />, div);
    ReactDOM.unmountComponentAtNode(div);
  })

  it('renders empty li', () => {
    const tree = renderer.create(<Accordion />).toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('The component renders no sections as active by default.', () => {
    const tree = renderer.create(<Accordion sections={[{ title: "Thing 1", content: "Lorem ipsum dolor sit amet." }]} />).toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('The component opens a clicked section.', () => {
    const wrapper = shallow(<Accordion />);
    wrapper.find('button').at(1).simulate('click');
    expect(toJson(wrapper)).toMatchSnapshot();
  })

  it('The component only opens the last section when multiple sections have been clicked.', () => {
    const wrapper = shallow(<Accordion />);
    wrapper.find('button').at(0).simulate('click');
    wrapper.find('button').at(1).simulate('click');
    expect(toJson(wrapper)).toMatchSnapshot();
  })
});
