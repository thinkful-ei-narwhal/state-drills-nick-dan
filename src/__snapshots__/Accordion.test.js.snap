// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordian Component The component only opens the last section when multiple sections have been clicked. 1`] = `
<ul
  className="container"
>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Thing 1
    </button>
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Thing 2
    </button>
    <p>
      This should appear and the others go away.
    </p>
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Thing 3
    </button>
  </li>
</ul>
`;

exports[`Accordian Component The component opens a clicked section. 1`] = `
<ul
  className="container"
>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Thing 1
    </button>
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Thing 2
    </button>
    <p>
      This should appear and the others go away.
    </p>
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Thing 3
    </button>
  </li>
</ul>
`;

exports[`Accordian Component The component renders no sections as active by default. 1`] = `
<ul
  className="container"
>
  <li>
    <button
      onClick={[Function]}
    >
      Thing 1
    </button>
  </li>
</ul>
`;

exports[`Accordian Component renders empty li 1`] = `
<ul
  className="container"
>
  <li>
    <button
      onClick={[Function]}
    >
      Thing 1
    </button>
  </li>
  <li>
    <button
      onClick={[Function]}
    >
      Thing 2
    </button>
  </li>
  <li>
    <button
      onClick={[Function]}
    >
      Thing 3
    </button>
  </li>
</ul>
`;
